packages/hardhat-chai-matchers/src/reverted/revertedWithCustomError.ts

```diff
@@ -0,0 +1,230 @@
+      // check the case where users forget to pass the contract as the first
+      // argument
+      if (typeof contract === "string" || contract?.interface === undefined) {
+        throw new TypeError(
+          "The first argument of .revertedWithCustomError has to be the contract that defines the custom error"
+        );
+      }
+
+      // validate custom error name
+      if (typeof expectedCustomErrorName !== "string") {
+        throw new TypeError(
+          "Expected a string as the expected custom error name"
+        );
+      }
+
+      const iface: any = contract.interface;
+
+      const expectedCustomError = findCustomErrorByName(
+        iface,
+        expectedCustomErrorName
+      );
+
+      // check that interface contains the given custom error
+      if (expectedCustomError === undefined) {
+        throw new AssertionError(
+          `The given contract doesn't have a custom error named ${expectedCustomErrorName}`
+        );
+      }
+
+      const onSuccess = () => {
+        this.assert(
+          false,
+          `Expected transaction to be reverted with custom error '${expectedCustomErrorName}', but it didn't revert`
+        );
+      };
+
+      const onError = (error: any) => {
+        const returnData = getReturnDataFromError(error);
+        const decodedReturnData = decodeReturnData(returnData);
+
+        if (decodedReturnData.kind === "Empty") {
+          this.assert(
+            false,
+            `Expected transaction to be reverted with custom error '${expectedCustomErrorName}', but it reverted without a reason string`
+          );
+        } else if (decodedReturnData.kind === "Error") {
+          this.assert(
+            false,
+            `Expected transaction to be reverted with custom error '${expectedCustomErrorName}', but it reverted with reason '${decodedReturnData.reason}'`
+          );
+        } else if (decodedReturnData.kind === "Panic") {
+          this.assert(
+            false,
+            `Expected transaction to be reverted with custom error '${expectedCustomErrorName}', but it reverted with panic code ${decodedReturnData.code.toHexString()} (${
+              decodedReturnData.description
+            })`
+          );
+        } else if (decodedReturnData.kind === "Custom") {
+          if (decodedReturnData.id === expectedCustomError.id) {
+            // add flag with the data needed for .withArgs
+            const customErrorAssertionData: CustomErrorAssertionData = {
+              contractInterface: iface,
+              customError: expectedCustomError,
+              returnData,
+            };
+            utils.flag(
+              this,
+              CUSTOM_ERROR_ASSERTION_DATA,
+              customErrorAssertionData
+            );
+
+            this.assert(
+              true,
+              null,
+              `Expected transaction NOT to be reverted with custom error '${expectedCustomErrorName}', but it did`
```

Comment:
Grammar nazi, checking in: mixed tense here. Should either be "NOT to be reverted...but it was" or "NOT to revert...but it did"

Reply:
Haha, love it, thanks. Fixed for all matchers in 027025d0586c496c889cabea34727b7fcb248db0